pkg/prometheus/resource_selector.go

```diff
@@ -677,6 +677,19 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			}
 		}
 
+		var scrapeInterval, scrapeTimeout monitoringv1.Duration = "", ""
+		if sc.Spec.ScrapeInterval != nil {
+			scrapeInterval = *sc.Spec.ScrapeInterval
+		}
+
+		if sc.Spec.ScrapeTimeout != nil {
+			scrapeTimeout = *sc.Spec.ScrapeTimeout
+		}
+
+		if err = validateScrapeIntervalAndTimeout(rs.p, scrapeInterval, scrapeTimeout); err != nil {
+			break
```

Comment:
```suggestion
			rejectFn(sc, err)
			continue
```

Reply:
should we update similar for other scrape interval and timeout checks as well? I can do that in separate PR

Reply:
Yes please. From what I can tell, there are a few places where invalid resources aren't effectively rejected (though logged as such). Case in point:

https://github.com/prometheus-operator/prometheus-operator/blob/c5ca520ea3b505c7fe12bb0a0eedccb46b0decc2/pkg/prometheus/resource_selector.go#L535-L548

If `validateRelabelConfig()` fails,  the probe will still be added to the final slice.